index.test-d.ts

```diff
@@ -3,35 +3,85 @@ import camelcaseKeys = require('.');
 const camelFooBarArray = camelcaseKeys(fooBarArray);
-expectType<typeof fooBarArray>(camelFooBarArray);
+expectType<Array<{fooBar: boolean}>>(camelFooBarArray);
 
-expectType<Array<{[key in 'foo-bar']: true}>>(camelcaseKeys([{'foo-bar': true}]));
+expectType<Array<{fooBar: boolean}>>(camelcaseKeys([{'foo-bar': true}]));
 
 expectType<string[]>(camelcaseKeys(['name 1', 'name 2']));
 
 expectType<string[]>(camelcaseKeys(['name 1', 'name 2'], {deep: true}));
 
-expectType<{[key in 'foo-bar']: true}>(camelcaseKeys({'foo-bar': true}));
+expectType<readonly [{ fooBar: true }, {fooBaz: true}]>(
+	camelcaseKeys([{'foo-bar': true}, {'foo-baz': true}] as const)
+);
 
-expectType<{[key in 'foo-bar']: true}>(
-	camelcaseKeys({'foo-bar': true}, {deep: true})
+expectType<{fooBar: boolean}>(camelcaseKeys({'foo-bar': true}));
+expectType<{fooBar: boolean}>(camelcaseKeys({'--foo-bar': true}));
+expectType<{fooBar: boolean}>(camelcaseKeys({foo_bar: true}));
+expectType<{fooBar: boolean}>(camelcaseKeys({'foo bar': true}));
+
+expectType<{fooBar: true}>(camelcaseKeys({'foo-bar': true} as const));
+expectType<{fooBar: true}>(camelcaseKeys({'--foo-bar': true} as const));
+expectType<{fooBar: true}>(camelcaseKeys({foo_bar: true} as const));
+expectType<{fooBar: true}>(camelcaseKeys({'foo bar': true} as const));
+
+expectType<{fooBar: {fooBar: {fooBar: boolean}}}>(
+	camelcaseKeys(
+		{'foo-bar': {foo_bar: {'foo bar': true}}},
+		{deep: true}
+	)
 );
 
-expectType<{[key in 'foo-bar']: true}>(
-	camelcaseKeys({'foo-bar': true}, {deep: true, pascalCase: true})
+expectType<{FooBar: boolean}>(
+	camelcaseKeys({'foo-bar': true}, {pascalCase: true})
+);
+expectType<{FooBar: true}>(
+	camelcaseKeys({'foo-bar': true} as const, {pascalCase: true})
+);
+expectType<{FooBar: boolean}>(
+	camelcaseKeys({'--foo-bar': true}, {pascalCase: true})
+);
+expectType<{FooBar: boolean}>(
+	camelcaseKeys({foo_bar: true}, {pascalCase: true})
+);
+expectType<{FooBar: boolean}>(
+	camelcaseKeys({'foo bar': true}, {pascalCase: true})
+);
+expectType<{FooBar: {FooBar: {FooBar: boolean}}}>(
+	camelcaseKeys(
+		{'foo-bar': {foo_bar: {'foo bar': true}}},
+		{deep: true, pascalCase: true}
+	)
 );
 
-expectType<{[key in 'foo-bar']: true}>(
-	camelcaseKeys({'foo-bar': true}, {exclude: ['foo', /bar/]})
+expectType<{fooBar: boolean; foo_bar: true}>(
+	camelcaseKeys(
+		{'foo-bar': true, foo_bar: true},
+		{exclude: ['foo', 'foo_bar', /bar/] as const}
+	)
 );
 
-expectType<{[key in 'foo-bar']: true}>(
+expectType<{fooBar: boolean}>(
 	camelcaseKeys({'foo-bar': true}, {stopPaths: ['foo']})
 );
+expectType<{topLevel: {fooBar: {'bar-baz': boolean}}; fooFoo: boolean}>(
+	camelcaseKeys(
+		{'top-level': {'foo-bar': {'bar-baz': true}}, 'foo-foo': true},
+		{deep: true, stopPaths: ['top-level.foo-bar'] as const}
+	)
+);
+
+expectType<Record<string, string>>(
+	camelcaseKeys({} as Record<string, string>)
+);
+
+expectType<Record<string, string>>(
+	camelcaseKeys({} as Record<string, string>, {deep: true})
+);
 
```

Comment:
Are you sure the tests here cover all the cases?

Reply:
Yes, I think, but you can suggest more cases.